packages/ses/src/module-load.js

```diff
@@ -109,9 +106,37 @@ export const load = async (
   // Await all dependencies to load, recursively.
   await Promise.all(
     values(resolvedImports).map(fullSpecifier =>
+      // Behold: recursion.
+      // eslint-disable-next-line no-use-before-define
       load(compartmentPrivateFields, moduleAliases, compartment, fullSpecifier),
     ),
   );
 
   return moduleRecord;
 };
+
+// `load` asynchronously loads `StaticModuleRecords` and creates a complete
+// graph of `ModuleCompartmentRecords`.
+// The module records refer to each other by a reference to the dependency's
+// compartment and the specifier of the module within its own compartment.
+// This graph is then ready to be synchronously linked and executed.
+export const load = async (
+  compartmentPrivateFields,
+  moduleAliases,
+  compartment,
+  moduleSpecifier,
+) => {
+  return loadWithoutErrorAnnotation(
+    compartmentPrivateFields,
+    moduleAliases,
+    compartment,
+    moduleSpecifier,
+  ).catch(error => {
+    const { name } = compartmentPrivateFields.get(compartment);
+    throw new Error(
```

Comment:
Given

https://github.com/Agoric/SES-shim/blob/cbe689e018b2786076d18c1883d75b00eb051fb4/packages/ses/src/tame-error-constructor.js#L9-L17

we should reproduce the error class as well

```suggestion
    const ErrorConstructor = NativeErrors[error.name] || Error;
    throw new ErrorConstructor(
```

Rather than importing `NativeErrors` from tame-error-constructor.js, it should go in whitelist.js. But that does not need to happen in this PR.

Reply:
Given the recursion, is the intent that each level of recursion adds its own location info?

Reply:
The problem with reconstructing the error object is that you silently lose the call stack associated with the original error object. What assumptions are you making about ambient availability of a `console` object for this code? Nothing I've done for keeping track of error stack correspondence is close to ready. One thing this code can do in the meantime is to do something like

```js
    const newError = NewErrorContructor(...);
    if (someAppropriateOption) {
      console.log(error, 'rethrown as', newError);
    }
    throw newError;
```

The normal console behavior is to emit the stacks associated with error objects it logs. This would at least give us the correspondences to grovel through to see the original error stack.

Reply:
Depending on your goals and your `console` assumptions, we could take the opposite approach:

```js
    if (someAppropriateOption) {
      console.log(error, `, loading  ${q(moduleSpecifier)} in compartment ${q(name,)}`);
    }
    throw error;
```

This completely avoids recreating new error objects as well as modifying the thrown one in place. It rethrows the one we've got, which retains the stack trace it was constructed with. All the new compartment-name diagnostic info in only logged to the `console`.

Reply:
Would it suffice to reüse the `error.constructor`?